File: lab2/task2.py
# Task2
n = 0
while n < 10 or n > 100:
    n = int(input("Please enter a number between 10 and 100: "))
    if n < 10 or n > 100:
        print("Invalid input. Please enter a number between 10 and 100: ")

fiz = 0
buz = 0
fizbuz = 0

for i in range(1, n + 1):

    if i % 7 == 0:
        print("(", i, "is skipped )")
        continue

    if i % 3 == 0 and i % 5 == 0:
        print("FizzBuzz")
        fizbuz = fizbuz + 1


    elif i % 3 == 0:
        print("Fizz")
        fiz = fiz + 1


    elif i % 3 != 0 and i % 5 == 0:
        print("Buzz")
        buz = buz + 1


    else:
        print(i)


print("--- Summary ---")
print("Fizz count :", fiz)
print("Buzz count :", buz)
print("FizzBuzz count:", fizbuz)

File: lab2/task2.cpp
#include <iostream>
using namespace std;

int main() {
    int n = 0;

    while (n < 10 || n > 100) {
        cout << "Please enter a number between 10 and 100: ";
        cin >> n;
        if (n < 10 || n > 100) {
            cout << "Invalid input. Please enter a number between 10 and 100: " << endl;
        }
    }

    int fiz = 0;
    int buz = 0;
    int fizbuz = 0;

    for (int i = 1; i <= n; i++) {
        if (i % 7 == 0) {
            cout << "(" << i << " is skipped )" << endl;
            continue;
        }

        if (i % 3 == 0 && i % 5 == 0) {
            cout << "FizzBuzz" << endl;
            fizbuz = fizbuz + 1;
        }
        else if (i % 3 == 0) {
            cout << "Fizz" << endl;
            fiz = fiz + 1;
        }
        else if (i % 5 == 0) {
            cout << "Buzz" << endl;
            buz = buz + 1;
        }
        else {
            cout << i << endl;
        }
    }

    cout << "--- Summary ---" << endl;
    cout << "Fizz count : " << fiz << endl;
    cout << "Buzz count : " << buz << endl;
    cout << "FizzBuzz count: " << fizbuz << endl;

    return 0;
}
